### PR TITLE
Stateless component and updated doc

### PR DESCRIPTION
**Pull request by yannickl88**
Updated the component to also include the stateless versions and updated doc to show the new cases

### CODE REVIEW

**Comment by nicoschoenmaker** [comment]
The new docs look good to me.

Perhaps a reference to the similarity of this design with the design of the form component can help people understand why it is this way.

Or perhaps do the before/after thing for the controller, so people can understand the gain.

**Comment by iltar** [comment]
@nicoschoenmaker perhaps a trimmed version of http://wouterj.nl/2016/08/trimming-your-controllers-using-form-handlers/ ?

**Comment by yannickl88** [comment]
I've already added a small note to the usage section:

> If you are familiar with the Symfony form component, these examples will seem very familiar. This is by design, the form handlers are designed to be symmetrical to the form types in the way they are used and configured.

**Comment by iltar** [comment]
# Preparations for 2.0

The current release is a preparation between the 1.x and 2.x versions. To ease the migration, you can already use the new structure as planned for 2.x, while still using the old setup. This is the latest 1.x feature release and we will not be patching bugs for the old 1.x methods. Version 2.0 will have the same feature set as 1.2, except for the legacy: `Hostnet\Component\Form\*`. Before you can use 2.0, you must update your code as explaining in [the migration guide](MIGRATION.md).

```
# Preparations for 2.0

The current release is a preparation between the 1.x and 2.x versions. To ease the migration, you can already use the new structure as planned for 2.x, while still using the old setup. This is the latest 1.x feature release and we will not be patching bugs for the old 1.x methods. Version 2.0 will have the same feature set as 1.2, except for the legacy: `Hostnet\Component\Form\*`. Before you can use 2.0, you must update your code as explaining in [the migration guide](MIGRATION.md).
```

**Comment by yannickl88** [comment]
@iltar add your bit of text in between installation and usage.

**Comment by iltar** [comment]
Some feedback I got from javiereguiluz:

> I expected something like ...
“This component makes it easier to process forms in Symfony applications"
or “improves the processsing of forms…"  
and then clearly explain why
what’s the trick?

So my suggestion added to this:
 - Reduce the readme.md to just an intro and some basic information about the package
 - Extract the docs to their own files (possibly in a docs directory)

This would reduce the chaos to maintain and read the current doc files and gives us the ability to properly link to each of them.

**Comment by yannickl88** [comment]
@iltar I see what you mean, but I think that is more of a github pages thing. I recommend making a different PR for that.

We can however try to improve the introduction. I'm open for suggestions :smiley:

**Comment by iltar** [comment]
What about something like this?

----

The form handlers are designed to enhance the developer experience (DX) when working with Symfony forms. It aims to reduce boilerplate code in the controller and move the success and failure flows to a separate class.

```php
class YourController extends Controller
{
    public function formAction(Request $request, MyEntityUser $user)
    {
        $handler = $this->get('hostnet.form_handler.factory')->create(MyFormHandler::class);
        
        if (($response = $handler->handle($request, new MyFormData())) instanceof RedirectResponse) {
            return $response;
        }
        // regular or in-valid flow
        return $this->render->renderView('/your/form.html.twig', [
            'form' => $handler->getForm()->createView()
        ]);
    }
}
```

By extracting the success -and if available, the failure- flows, you reduce the amount of code in your controllers, which in turn, achieves slim controllers. The definition of a controller is according to Symfony: _["a PHP function you create that reads information from the Symfony's Request object and creates and returns a Response object"](http://symfony.com/doc/current/controller.html)_.

**Comment by javiereguiluz** [comment]
What about changing:

```
It aims to reduce boilerplate code in the controller ...
```

by:

```
It reduces boilerplate code in the controller ...
```

or even:

```
It makes the controllers simpler by moving the form success and failure flows to a separate class.
```

**Comment by javiereguiluz** [comment]
In any case, the proposed intro is much better in my opinion!! Now it's easy to understand the purpose of this code and why should I use it. Thanks!

**Comment by yannickl88** [comment]
Thanks @javiereguiluz I've updated the PR (and the branch) with your suggestion.

File: LEGACY.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+form-handler-bundle
+===================
+> This documentation is for the `1.*` Form Handler architecture. It is here for legacy reasons.
+
+The form handler bundle is designed give extra support on the [form handler component](https://github.com/hostnet/form-handler-component). This includes predefined services like the `SimpleFormProvider` and a handy `ParamConverter` to easily inject your form handler into a controller.
```
Comment (iltar) [inline LEGACY.md]: What about: 
```
The form handler bundle is designed integrate the form handler component into your Symfony application.
```

**Pull request merged**